src/Api/include/Api/EncodedString.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_API_ENCODED_STRING_H_
+#define ORBIT_API_ENCODED_STRING_H_
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+
+// In order to avoid expensive allocations in proto buffers, we encode the first 64 characters in
+// eight 64bit fields (byte by byte). Any additional characters will be also encoded in a sequence
+// of 64bit fields. So a string [0x11, 0x22, 0x33] would get encoded as 0x0000000000332211, all
+// other fields are zero and there are no additional fields.
+namespace orbit_api {
+
+template <typename Dest>
+void EncodeString(const char* source, Dest* dest, void (Dest::*write_chunk_1)(uint64_t),
```

Comment:
It would be nice if this worked for any number of chunks. If we want to change the number of chunks, let's say after a statistical analysis to optimize the size of protos, then we need to change the interface and the implementation. Also, the current implementation can't be reused by other systems where the string size would be typically different and thus might require a different number of fixed chunks.

Maybe use std::array<uint64_t> and use the number of chunks as template argument for the fixed part? In the implementation we could then use a loop instead of copying code blocks 8 times.

Reply:
Good idea. In order to move a bit faster, I'd suggest to keep this for now. I created b/195377283 and assigned it to me, in order to keep track of this. I might tackle this next week.